libp2p/upgrademngrs/muxedupgrade.nim

```diff
@@ -62,14 +62,14 @@ proc mux*(
   muxer.handler = muxer.handle()
   return muxer
 
-proc upgrade(
+method upgrade*(
   self: MuxedUpgrade,
   conn: Connection,
   direction: Direction,
   peerId: Opt[PeerId]): Future[Muxer] {.async.} =
   trace "Upgrading connection", conn, direction
 
-  let sconn = await self.secure(conn, direction, peerId) # secure the connection
+  let sconn = await self.secure(conn, direction, if direction == In: Opt.none(PeerId) else: peerId) # secure the connection
```

Comment:
Why is this necessary?

Reply:
`upgradeIncoming` did this `self.upgrade(conn, In, Opt.none(PeerId))`

Reply:
I think we can just forward the PeerId, it will be `Some` during HP, and will actually improve security by checking the peer identity earlier (are you checking peer identity btw currently?)

Reply:
AFAIK I'm not checking it. Could you please elaborate more on that?

Reply:
I meant checking that the connection established during HP is indeed with the correct peer (by checking the peerid)

Reply:
In this PR we're always giving the remote peer id to the upgrade process.